package image;

message ClassifyRequest {
    enum RequestType {
        CLASSIFY = 0;
        SIMILARITY = 1;
        SEARCH = 2;
    }
    required bytes file_name = 1;
    optional int32 top_n_result = 2 [default = 5];
    optional bytes compare_file_name = 3;
    optional RequestType request_type = 4 [default = CLASSIFY];
    optional float min_sim_thres = 5 [default = 0.25];
    optional float max_sim_thres = 6 [default = 1.0];
    optional int32 feature_layer = 7 [default = 20];
}


message ClassifyResult {
    optional bytes category_name = 1;
    optional int32 category_id = 2;
    optional float category_weight = 3;
}
message SearchResult {
    optional bytes search_file_name = 1;
    optional float search_similarity = 2;
}
message ClassifyResponse {
    repeated ClassifyResult rsp_res = 1;
    optional float similarity = 2;
    repeated SearchResult search_res = 3;
}
